midend/interpreter.cpp

```diff
@@ -310,33 +310,45 @@ void SymbolicStruct::dbprint(std::ostream& out) const {
 }
 
 SymbolicHeaderUnion::SymbolicHeaderUnion(const IR::Type_HeaderUnion* type,
-                               bool uninitialized,
-                               const SymbolicValueFactory* factory) :
-        SymbolicStruct(type, uninitialized, factory),
-        valid(new SymbolicBool(false)) {}
-
-void SymbolicHeaderUnion::setValid(bool v) {
-    if (!v)
-        setAllUnknown();
-    valid = new SymbolicBool(v);
+                                         bool uninitialized,
+                                         const SymbolicValueFactory* factory) :
+        SymbolicStruct(type, uninitialized, factory) {}
+
+SymbolicBool* SymbolicHeaderUnion::isValid() const {
+    int validFields = 0;
```

Comment:
This is necessary to calculate if `HeaderUnion `has more than one valid field

Reply:
If I understand your statement correctly (perhaps I do not), according to the P4_16 language specification, it is guaranteed that at most one member of a header union is valid at a time.  If member m1 of a header union is valid, and later you make a different member m2 valid, then automatically at the same time that member m2 becomes valid, m1 must become invalid.

Reply:
Indeed, below he calls BUG if this happens.

Reply:
But this function doesn't validate anyone, it just evaluates valid for HeaderUnion.

Reply:
I probably understood that when I call setValid for Header from HeaderUnion, I need to check that HeaderUnion is not valid now, if valid make it setAllUnknown and then set valid for Header